Add an all method

DebbieGillespie opened the pull request:
Adds an all method to return all rows from a table.  Also includes some minor changes to the test to remove duplicated code and improve variable names.

Commit:
Adds a .ruby-version file

Need to make sure using ruby 2.2.0 otherwise the tests won't run.

Commit:
Removes the duplicated persistance_class assignment

persistance_class is already getting set at the top of the spec file -
don't need to set it a second time.

Commit:
Use the provided campus_repository

A campus repository is already defined, so better to use it then
redefine it.

Commit:
Adds an all method

Need a way to return all entities for a repository.

Still need to add tests - having trouble getting the tests to work, but
want to commit what I do have that is working.

Commit:
Change campus_repository to repository

It was not actually representing a campus, so campus_repository was a
misleading name.

Commit:
Call map_record_out on each entity

Need to call map_record_out on each entity that the all method returns

Commit:
Adds tests to test the all method

Commit:
Return version back to 0.0.1

It is not necessary to change the version number at this time.

IanWhitney commented:
+1. We might want to refactor those nested method calls in the repository sometime, but I'm ok with it for now.

davinlagerroos commented on spec/lib/hexagram/repositories/repository_spec.rb:
```diff
@@ -111,19 +111,55 @@
+
+  describe "all" do
+    before do
+      @rand_value = rand
```
Is there a reason to use instance variables instead of 'let's? let takes a block, so I think you can do multi-line setup as long as the last line contains the intended return value

DebbieGillespie replied:
I tried to use let instead, but I get the following error:

`let` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block).

IanWhitney replied:
I'm also remembering that using let with `rand` leads to the variable returning a different number each time you call it. But that may no longer be true. I'll check.

davinlagerroos commented on lib/hexagram/repositories/repository.rb:
```diff
@@ -11,6 +11,10 @@ def find(id)
         map_record_out(orm_adapter.find(id, persistence_class))
       end
 
+      def all()
```
If a method doesn't take parameters, convention is to leave off the parenthesis https://github.com/bbatsov/ruby-style-guide#method-parens

Commit:
Remove parenthesis from method definition

Ruby convention leaves off parenthesis if there are no parameters for
the method

whit0694 committed:
Convert test double to set id & value from attributes

whit0694 committed:
convert test to use lets

Commit:
Removes the calls to :focus

Don't want this in the repo - only useful during development

Commit:
Merge pull request #2 from umn-asr/pr/1

Test cleanup
